devlab/tests/functional_test.py

```diff
@@ -60,10 +60,15 @@ def filter_networks(self):
     # 'name' parameter in it's detail.
     # Implimentation of Subnet filtering for no name subnets is still needed.
     def filter_subnets(self):
-        subnets = [i['name'] for i in config.subnets]
-        for i in config.tenants:
-            if 'subnets' in i:
-                for subnet in i['subnets']:
+        subnets = [i['name'] for net in config.networks if net.get('subnets')
+                   for i in net['subnets']]
+        for tenant in config.tenants:
+            if 'networks' not in tenant:
+                continue
+            for network in tenant['networks']:
+                if 'subnets' not in network:
+                    continue
+                for subnet in network['subnets']:
```

Comment:
The main issues here - subnet name can be empty... So maybe you should think about how to filter using other parameter.